exec/java-exec/src/test/java/org/apache/drill/exec/store/log/TestLogReader.java

```diff
@@ -177,6 +180,15 @@ private static LogFormatConfig firewallConfig() {
         regex, "ssdlog", null, schema);
   }
 
+  // DRILL-7853
+  private static LogFormatConfig issue7853Config() {
+    String regex = "([^ ]*) ([^ ]*) ([^ ]*) ([^ ]*):([0-9]*) ([^ ]*)[:-]([0-9]*) ([-.0-9]*) ([-.0-9]*) ([-.0-9]*) (|[-0-9]*) (-|[-0-9]*) ([-0-9]*) ([-0-9]*) \\\"([^ ]*) ([^ ]*) (- |[^ ]*)\\\" \\\"([^\\\"]*)\\\" ([A-Z0-9-]+) ([A-Za-z0-9.-]*) ([^ ]*) \\\"([^\\\"]*)\\\" \\\"([^\\\"]*)\\\" \\\"([^\\\"]*)\\\" ([-.0-9]*) ([^ ]*) \\\"([^\\\"]*)\\\"($| \\\"[^ ]*\\\")(.*)";
```

Comment:
Nit: to make the pattern more readable, use `\d*` in place of `[0-9]*` and `\W*` in place of `[(^ ]*)`.

One of the problems with the log format reader is that the above long regex is nearly unreadable. Better, in code, to provide the poor reader some hints (as we do elsewhere):

```
   regex = "\\W*"  // field 1 name
             + " \\W*" // field 2 name
             ...
```

Or, even better:

```
    String plain = "\\W*";
    String integer = "\\d*";
    String sep = " ";
    ...
    regex = plain // field1
              + sep + plain // field 2
     ...
```

Not pretty, but makes the code more readable. And, as you create the format, you can comment out all but a few fields.

For production use, there should be some simple way to provide this in the config since huge amount of time will be wasted trying to get the long regex correct in the current design.

Reply:
Even better, whittle this test, and the required log file, down to the one field which failed: the date field. Then, the code here, and the test code below, can simply verify that the log reader is correctly parsing the date field (using Java date/time formats.)

If you make the log file itself simpler, then you can write the log file in the test. See the various `TestCsv*` tests for an example of how to do this.